app/kubernetes-ingress-controller/1.1.x/references/annotations.md

```diff
@@ -0,0 +1,445 @@
+
+> Available since controller 0.8
+
+This annotation can be applied to an Ingress resource and can take two values:
+- `"true"`: If set to true, the part of the path specified in the Ingress rule
+  will be stripped out before the request is sent to the service.
+  For example, if the Ingress rule has a path of `/foo` and the HTTP request
+  that matches the Ingress rule has the path `/foo/bar/something`, then
+  the request sent to the Kubernetes service will have the path
+  `/bar/something`.
+- `"false"`: If set to false, no path manipulation is performed.
+
+All other values are ignored.
+Please note the quotes (`"`) around the boolean value.
+
+Sample usage:
+
+```yaml
+konghq.com/strip-path: "true"
+```
+
+### konghq.com/preserve-host
+
+> Available since controller 0.8
+
+This annotation can be applied to an Ingress resource and can take two values:
+- `"true"`: If set to true, the `host` header of the request will be sent
+  as is to the Service in Kubernetes.
+- `"false"`: If set to false, the `host` header of the request is not preserved.
+
+Please note the quotes (`"`) around the boolean value.
+
+Sample usage:
+
+```yaml
+konghq.com/preserve-host: "true"
+```
+
+### konghq.com/https-redirect-status-code
+
+> Available since controller 0.8
+
+By default, Kong sends HTTP Status Code 426 for requests
+that need to be redirected to HTTPS.
+This can be changed using this annotations.
+Acceptable values are:
+- 301
+- 302
+- 307
+- 308
+- 426
+
+Any other value will be ignored.
+
+Sample usage:
+
+```yaml
+konghq.com/https-redirect-status-code: "301"
+```
+
+Please note the quotes (`"`) around the integer value.
+
+### konghq.com/regex-priority
+
+> Available since controller 0.9
+
+Sets the `regex_priority` setting to this value on the Kong route associated
+with the Ingress resource. This controls the [matching evaluation
+order](/../../latest/proxy/#evaluation-order) for regex-based
+routes. It accepts any integer value. Routes are evaluated in order of highest
+priority to lowest.
+
+Sample usage:
+
+```yaml
+konghq.com/regex-priority: "10"
+```
+
+Please note the quotes (`"`) around the integer value.
+
+### konghq.com/methods
+
+> Available since controller 0.9
+
+Sets the `methods` setting on the Kong route associated with the Ingress
+resource. This controls which request methods will match the route. Any
+uppercase alpha ASCII string is accepted, though most users will use only
+[standard methods](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods).
+
+Sample usage:
+
+```yaml
+konghq.com/methods: "GET,POST"
+```
+
+### konghq.com/snis
+
+> Available since controller 1.1
+
+Sets the `snis` match criteria on the Kong route associated with this Ingress.
+These allow route matching based on the server name indication information sent
+in a client's TLS handshake, for route-attached plugins that execute during the
+certificate phase (for example, [Mutual TLS
+Authentication](https://docs.konghq.com/hub/kong-inc/mtls-auth/)).
```

Comment:
```suggestion
When using route-attached plugins that execute during the certificate 
phase (for example, [Mutual TLS Authentication](/hub/kong-inc/mtls-auth/)), 
the `snis` annotation allows route matching based on the server name 
indication information sent in a client's TLS handshake.
```
Context first, if possible, so the reader knows if they need to keep reading or not.